hazelcast-openshift-rhel/healthcheck.sh

```diff
@@ -1,3 +1,14 @@
-#!/bin/bash                                                                     
-result=$( curl http://127.0.0.1:5701/hazelcast/healthcheck | awk 'match($0, /Hazelcast\:\:ClusterSafe\=([A-Z]*)/)' )                                            
-if [ "$result" == "Hazelcast::ClusterSafe=TRUE" ]; then exit 0; else exit 1; fi 
+#!/bin/bash
+URL="http://127.0.0.1:5701/hazelcast/healthcheck"
+
+HTTP_RESPONSE=$(curl --silent --write-out "HTTPSTATUS:%{http_code}" $URL)
+HTTP_BODY=$(echo $HTTP_RESPONSE | sed -e 's/HTTPSTATUS\:.*//g')
+HTTP_STATUS=$(echo $HTTP_RESPONSE | tr -d '\n' | sed -e 's/.*HTTPSTATUS://')
+
+if [ ! $HTTP_STATUS -eq 200  ]; then
+  echo "failure, http status: "$HTTP_STATUS
+  exit 1
+fi
+
+result=$(echo $HTTP_BODY | awk '{match($0,  /Hazelcast\:\:NodeState\=([A-Z]*)/, m)}END{print m[1]}')
```

Comment:
Nope we need the `clustersafe` state because otherwise there are still ongoing migration which would cause data loss when a node is restarted / shutdown too fast (like scale down or rolling update).